### ISSUE TITLE
Support for BorshDeserialize and BorshSerialize

### ISSUE THREAD

**Issue by ivanceras** [paritytech/parity-common#640]
Hi, is there any plan to support BorshSerialize and BorshDeserialize for the ethereum-types.
I notice that borsh is used in NEAR and in Solana which provides compatibility with ethereum.

**Comment by ordian** [comment]
Hey @ivanceras. If there's enough demand for it, we might consider adding this feature. However, that would mean tying any major releases to a breaking change to ethereum-types, which we'd like to avoid.

**Comment by sloboste** [comment]
Ah, didn't see that issue - thanks for pointing it out @ordian. 

I'll drop a comment in https://github.com/paritytech/parity-common/pull/642

**Comment by sloboste** [comment]
Borsh support would be useful to [hyperlane](https://www.hyperlane.xyz/) and [eclipse](https://twitter.com/eclipsefnd).

@ordian to address your concern about avoiding being tied to breaking changes - given that `borsh` is still in `0.x`, what do you think about feature gating borsh support behind an `experimental-impl-borsh` feature and documenting explicitly that there may be breaking changes to anything related to borsh?

**Comment by ordian** [comment]
@sloboste hey, sorry for the long delay in reply. I think your suggestion is a reasonable workaround we can live with.

**Comment by tkporter** [comment]
@ordian would love to see https://github.com/paritytech/parity-common/pull/698 get merged with the feature gate  :) are there any changes you'd like to see there?

**Comment by ordian** [comment]
@tkporter yes, see point 2 in https://github.com/paritytech/parity-common/pull/698#issuecomment-1331827459. Instead of introducing a new crate with a regular macro, we could just use derive with a feature gate.
